packages/astro/src/core/build/index.ts

```diff
@@ -144,7 +144,10 @@ class AstroBuilder {
 		);
 		await runHookConfigDone({ settings: this.settings, logger: logger });
 
+		const { ensureCodegenDirExists } = await import('../../config/types.js')
```

Comment:
It's weird that a function that contains **logic** inside a file is called `types`. Usually, a `types` file contains only... types :D

Reply:
Ahah right! The place was originally suggested by Erika but things changed quite a bit! If you have any suggestion on where to put those, let me know! The repo is so big that I struggled to find the most sensible place

Reply:
I'm not good at names. Maybe `typeGen.ts`?

Reply:
Still under `config`?

Reply:
Yeah I think so